rfc/text/advanced/ap_09_rpc_progressive_call_results.md

```diff
@@ -289,28 +289,16 @@ The progressive `YIELD` and progressive `RESULT` may also be empty, e.g. when th
 
 Even if a *Caller* has indicated its expectation to receive progressive results by setting `CALL.Options.receive_progress|bool := true`, a *Callee* is **not required** to produce progressive results. `CALL.Options.receive_progress` and `INVOCATION.Details.receive_progress` are simply indications that the *Caller* is prepared to process progressive results, should there be any produced. In other words, *Callees* are free to ignore such `receive_progress` hints at any time.
 
-**Progressive Call Result Cancellation**
+**Caller Leaving**
 
-Upon receiving a `YIELD` message from a *Callee* with `YIELD.Options.progress == true` (for a call that is still ongoing), if the original *Caller* is no longer available (has left the realm), then the *Dealer* will send an `INTERRUPT` to the *Callee*.  The `INTERRUPT` will have `Options.mode` set to `"killnowait"` to indicate to the client that no response should be sent to the `INTERRUPT`. This `INTERRUPT` is only sent in response to a progressive `YIELD` (`Details.progress == true`), and is not sent in response to a normal or final `YIELD`.
-```
-[INTERRUPT, INVOCATION.Request|id, Options|dict]
-```
-Options:
-```
-INTERRUPT.Options.mode|string == "killnowait"
-```
-
-Cancellation of Progressive Call Results addresses a potential security vulnerability: In cases where progressive results are used to stream data to *Callers*, and network connectivity is unreliable, *Callers* may often get disconnected in the middle of receiving such progressive results. Without the mandated cancellation behavior, recurring connect-call-disconnect cycles by a *Caller* would result in a rapidly growing backlog of unprocessed streaming results, overloading the router and further degrading network connectivity.
-
-The message flow for progressive results cancellation involves:
+The *Dealer*'s behavior for when a *Caller* leaves or disconnects during a progressive results call shall be the same as in a basic, non-progressive call. That is, the *Dealer* sends an INTERRUPT to the *Callee* with `mode="killnowait"`. See [Caller Leaving During RPC Invocation] (#rpc-caller-leaving) under the Basic Profile.
```

Comment:
Maybe its worth noting that if Callee had time to send a YIELD message before it received INTERRUPT from the Dealer, the DEALER has nothing to do with the YIELD message and just omits it.

Reply:
sorry, missed that.